storage/badger/payloads_test.go

```diff
@@ -23,7 +24,7 @@ func TestPayloadStoreRetrieve(t *testing.T) {
 		guarantees := badgerstorage.NewGuarantees(metrics, db)
 		results := badgerstorage.NewExecutionResults(metrics, db)
 		receipts := badgerstorage.NewExecutionReceipts(metrics, db, results)
-		store := badgerstorage.NewPayloads(db, index, guarantees, seals, receipts)
+		store := badgerstorage.NewPayloads(db, index, guarantees, seals, receipts, results)
 
 		blockID := unittest.IdentifierFixture()
 		expected := unittest.PayloadFixture(unittest.WithAllTheFixins)
```

Comment:
Would be great to have a test that verifies that blocks with arbitrary order of receipts and results are retrieved correctly. This only has a single result and receipt, which is insufficient to test that the order is preserved.
Would be great to have 10 results and receipts in arbitrary order.

Reply:
I've modified `WithAllTheFixins` to have 10 receipts, this adds extra receipts for test cases in `payloads_test.go` and `block_test.go` which test this scenario.